Geospatial.md

```diff
@@ -94,6 +94,41 @@ Bounding box is defined as the thrift struct below in the representation of
 min/max value pair of coordinates from each axis. Note that X and Y Values are
 always present. Z and M are omitted for 2D geospatial instances.
 
+Writers should follow the guidelines below when calculating bounding boxes in
+the presence of edge cases.
+
+* `null` instance: Skip it and continue processing the remaining 
+  geospatial instances. Do not produce a bounding box if all instances are null.
+* Non-`null` instance with [special geospatial values](#special-geospatial-values):
+  * X and Y: Skip any special X or Y value and continue processing the 
```

Comment:
I have a question here, do we mean:  skip any geospatial instance with special X and Y value

(ie, if a linestring has a thousand coordinates, but only one has bad X,Y value, its enough to skip whole instance?)

Reply:
no, what we mean is: if a linestring has a thousand coordinates and one has a bad Y value, we skip this single Y value, continue aggregating on other Y values in this linestring instance

Reply:
ok.  sorry i have another confusion in the doc, I know we define special geospatial value for the geospatial instance.  But then in the language we say skip special X or Y value, we dont define what is a special coordinate

Reply:
Taking a LineString object as an example `LINESTRING (0 NaN, 0 1, 1 2)`:

We have 3 levels here:

1. A geospatial instance: this linestring object
2. A geospatial coordinate:
	* we have 3 geospatial coordinates `0 NaN` , `0 1`, `1 2`
3. A geospatial value:
	* For X, we have `0`, `0`, `1`
	* For Y, we have `NaN`, `1`, `2`

The special geospatial value in this spec refers to the 3rd level. When calculating bbox, we only omitted that single bad value `NaN`, and use X (`0`, `0`, `1`) and Y (`1`, `2`).

That's why we didn't have a definition for `special coordinate` (2nd level) because we look at the 3rd level directly

Reply:
OK sorry i mix levels 2 and 3 in my head (coordinate and value).  

But currently the language of 'special geospatial value' from the link mentions things like 'Nan (point)' , 'empty geometry', and 'invalid WKB' which seems to talk about invalid instances (level 1), which is my confusion.  Then the section talks not about those, but about if actual x,v,z,m values are invalid.  Individual value is invalid only if NaN and outside the range (depending on context), right?

Reply:
@szehon-ho I think you’re right. The current wording conflated axis values and coordinates. I’ve updated the special geospatial value section to clarify the distinction.